### PR TITLE
DROOLS-4905 / DROOLS-4742: [DMN Designer] Code Completion - Index all elements named by users / Add suggestions about "requirements" and/or "dependencies" of a decision node

### PR DESCRIPTION
**Pull request by karreiro**
See:
- https://issues.redhat.com/browse/DROOLS-4905
- https://issues.redhat.com/browse/DROOLS-4742

### CODE REVIEW

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by jomarko** [comment]
jenkins execute full downstream build

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
(rebasing the branch :eyes:)

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
(rebasing the branch 👀)

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
@jomarko Should we re-assign this PR to Anna?

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by jomarko** [comment]
jenkins execute full downstream build

**Comment by jomarko** [comment]
- [ ] 20 nodes and 20 data types
- [ ] empty names
- [ ] long names
- [ ] multichar names
- [ ] Chrome & Firefox
- [ ] Imported nodes
- [ ] Imported types

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/codecompletion/MonacoFEELVariableSuggestions.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import org.kie.workbench.common.dmn.api.definition.model.DMNDiagram;
+import org.kie.workbench.common.dmn.api.definition.model.NamedElement;
+import org.kie.workbench.common.dmn.api.property.dmn.Name;
+import org.kie.workbench.common.dmn.client.graph.DMNGraphUtils;
+import org.kie.workbench.common.stunner.core.graph.Node;
+import org.kie.workbench.common.stunner.core.graph.content.definition.Definition;
+
+@ApplicationScoped
+public class MonacoFEELVariableSuggestions {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/codecompletion/MonacoFEELVariableSuggestions.java]: Are also data type variables? If not we should maybe rename. Also thinking if we shouldn't aply single responsibility principle and have one method `getVariableSuggestions` and `getDataTypesSuggestions`. What do you think?

**Comment by jomarko** [comment]
### Manual review 01
Found this [scenario](https://issues.redhat.com/secure/attachment/12464471/pick-option-move-to-other-grid.webm). When picked Variable and moved to other grid.

**Comment by jomarko** [comment]
### Manual review 02
Suggestions can not be canceled by **Esc**

- Start edit cell
- Press **Ctrl+Space**
- Press **Esc** - suggestions are not cancelled

**Comment by jomarko** [comment]
### Manual review note 03
If as 
Screenshot from 2020-01-30 14-12-58
data type so node has same name, we will get the suggestion twice

**Comment by jomarko** [comment]
### Manual review 04
Imported nodes are not offered. Imported types are.

Done with review.

**Comment by karreiro** [comment]
@jomarko Thanks for the review.

**Manual review 01**
Thanks for reporting :-) however, it wasn't introduced by this PR. Thus, I've raised this JIRA (so we can handle it separately): https://issues.redhat.com/browse/DROOLS-5006.

**Manual review 02**
Thanks for reporting! But the issue wasn't introduced by this PR. Thus, I've raised this JIRA (so we can handle it separately): https://issues.redhat.com/browse/DROOLS-5005 (btw, I've noticed it works when there's some text in the input field):


**Manual review note 03**
Fixed.

**Manual review 04**
Actually, it works for included nodes as well. Monaco shows a limited number of results, but if you type the included model name, you'll be able to see included nodes:

**Comment by jomarko** [comment]
@karreiro did you push? Github do not show changes since my last review

**Comment by karreiro** [comment]
@jomarko Probably a Github issue 👀 I've pushed it "again".

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/codecompletion/MonacoFEELVariableSuggestions.java (hunk 1 above)
Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/codecompletion/MonacoFEELVariableSuggestions.java]: Seems like a good idea. Applied!

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
(rebasing the branch... 👀)

**Comment by karreiro** [comment]
Jenkins please retest this.

**Pull request merged**